### DISCUSSION TITLE
Pokémon Database Scripts

### DISCUSSION THREAD

**Discussion by tylerjamesf** [tylerjamesf/MissingNo#9] [Ideas]
@phantomfirefox - Yeah I have thoughts, depending on how far down the rabbit hole you wanted to go. 

I am curious on your thoughts on "generations". As the games have evolved, see what I did there, it adds complexity or simple changes. i.e. Generation 1 (RBY) on Route 1 only had Pidgey & Rattata, but Gen 2 (GS) there are 3 new Pokémon plus the original two. I

One approach could be adding `region_id` to [route], but from what I could tell Route # was fairly redundant, i.e. most Regions appear to have the same Route 1 to 25. I was thinking about creating an a Region_Route_Xref where it would store the respective keys. But then curious if we need to maintain that the routes are different from one Generation to the next? (assuming routes get added / removed over time).

Same with Spawn Rates -- In Gen 1 it was 50/50, but Gen 2 we have 10 different splits to potentially account for and that only expands as we add more Gens. 

Sooooo we could account for those complexities now and only handle Gen 1, or keep it simple now, and then refactor the model if new Generations are added.  However, I don't really think about the Pokémon Universe in terms of Generations except when they were technically added. Unfortunately, it might require us to modify the data if we wanted to try and blend it all together for certain aspects (spawn / routes / etc.). If we keep them separated by Generation, it keeps it consistent with the game and while I assume it could help with Analytics it does add complexities there too.

_Originally posted by @evokercodex in https://github.com/phantomfirefox/MissingNo/pull/8#issuecomment-1095102645_

**Comment by tylerjamesf** [comment]
@evokercodex I think we should think about the bigger picture now on data and the database we want to construct rather that trying to reconstruct as we go along - I think that would get us into trouble.

I think we could look at it in two way:
1. Are we basing it of different regions for that Generation?
2. Are we basing it off regions and generation but per game structure i.e. Kanto would be red , blue, yellow, fireRed, leaf green, etc anywhere when a game brings your to that region?

Personally I would argue its the latter because the objective would be based around game date.; And omitting a game would be biased to he data.

What do you think?

**Reply by evokercodex** [reply]
Okay I think we are on the same page then as in we want to mimic the game's model (#2) vs trying to hodge podge something together. I can continue drafting up what that might look like for us. Hold tight!
